### PR TITLE
Don't color diff output when stdout is not a tty.

### LINKED ISSUE #435: Don't color when piping
Currently, rustfmt colors diffs even when piping. This leads to unreadable diffs when piping to a pager.

### PR DESCRIPTION
**Pull request by cassiersg**
Fixes #435

### CODE REVIEW

File: src/rustfmt_diff.rs [hunk 1]
```diff
@@ -88,6 +101,7 @@ pub fn print_diff<F>(diff: Vec<Mismatch>, get_section_title: F)
     where F: Fn(u32) -> String
 {
     let mut t = term::stdout().unwrap();
+    let isatty = unsafe { libc::isatty(libc::STDOUT_FILENO) } != 0;
```
Comment (nrc) [inline src/rustfmt_diff.rs]: Can we use something in libterm for this? `supports_attr` maybe? Seems kind of wrong to need a libc call here. If its not possible you should file an issue against libterm.

File: src/rustfmt_diff.rs [hunk 2]
```diff
@@ -96,19 +110,27 @@ pub fn print_diff<F>(diff: Vec<Mismatch>, get_section_title: F)
         for line in mismatch.lines {
             match line {
                 DiffLine::Context(ref str) => {
-                    t.reset().unwrap();
+                    if isatty {
+                        t.reset().unwrap();
+                    }
```
Comment (nrc) [inline src/rustfmt_diff.rs]: We should abstract stdout by wrapping it in an enum. We could put the isatty check in the constructor and encapsulate all these `if` expressions in methods.

File: src/rustfmt_diff.rs (hunk 1 above)
Comment (kamalmarhubi) [inline src/rustfmt_diff.rs]: fwiw, rustc does this for unix: https://github.com/rust-lang/rust/blob/f12d32d307d53a8f7851ce8dd1b4fbd7a4f0b50a/src/libsyntax/errors/emitter.rs#L659

**Comment by marcusklaas** [comment]
Superceded by https://github.com/rust-lang-nursery/rustfmt/pull/1023.

**Pull request closed without merging**